### DISCUSSION TITLE
Is there a way to make Jukebox reply to another bot's message?

### DISCUSSION THREAD

**Discussion by KTSnowy** [Hazmi35/jukebox#827] [Q&A] [answered]
For example, I wanted to have another bot send a message ("jukebox play *url*") and jukebox would execute the command the other bot sent.

Basically what I'm trying to do is have a GUI app with speech recognition sending these commands to discord through another bot, and then have jukebox reply to these commands.

**Comment by Hazmi35** [comment, marked as answer]
You can, but I generally avoid the idea of listening to bot (including own) messages, because it can cause security issues, for example, moderation commands usually check if the user has specific permissions, but when the bot has it and if non-moderators can use the bot to say something, that is a security issue.

But if you insist, you can  change this statement in https://github.com/Hazmi35/jukebox/blob/c77acc1d53cf548222b8691d9fba4c325c5527db/src/events/MessageCreateEvent.ts#L9

to 
```js
// If the message is bot's own message, return
if (message.author.id !== message.client.user.id || message.channel.type !== "GUILD_TEXT") return message; 
```
Or even better, you can exclude this filter to specific bots
```js
// If the user is a bot, and not some specific user, return
 if ((message.author.bot && message.author.id !== "SOME-ID") || message.channel.type !== "GUILD_TEXT") return message; 
```

**Reply by KTSnowy** [reply]
I think it would be possible to mitigate the security issue by only allowing the bot to reply to a specific bot like the example you gave and by only programming a limited set to commands in the speech recognition bot (so it can't send moderation commands)

Also, I'm curious about a thing in the above code

`message.channel.type !== "GUILD_TEXT"`

Does this make it so that the bot can't reply to DMs?

**Reply by Hazmi35** [reply]
Yes, because I assume voice-related stuff would break in DMs, but I haven't tested it

**Reply by KTSnowy** [reply]
It should be safe if the bot doesn't have any moderation commands right?

**Reply by Hazmi35** [reply]
As long as you're only listening to specific bots and it is authorized, yes

**Reply by KTSnowy** [reply]
Sorry if I'm being annoying, but one last question.
Is it possible for the bot to reply to webhook messagens?

**Reply by Hazmi35** [reply]
Uh not sure, I haven't tried doing that, try to log message variable before if statements, see if the message from webhook is received.

If it's received then find a way to identify the message is from that specific webhook for the security stuffs I said earlier

**Reply by Hazmi35** [reply]
Hi, does the issue solved? if yes, please mark this discussion as resolved, let me know if you had other questions
